### ISSUE TITLE
 Problem with getExternalCellsTotal

### ISSUE THREAD

**Issue by pordyna** [ComputationalRadiationPhysics/picongpu#3957]
I reused the handling of the new particle boundary offset from `particleCalorimeter` and noticed the following:

In the `ParticleCalorimeter.kernel` 
https://github.com/ComputationalRadiationPhysics/picongpu/blob/0655b4426eac1122e23fb580315f867f7253c699/include/picongpu/plugins/particleCalorimeter/ParticleCalorimeter.kernel#L112-L113
the boundaries are used in such a way that the `beginCellIdx` is the first cell of the valid range and `endCellIdx` is the first cell beyond this valid range. This intention is confirmed by this comment
https://github.com/ComputationalRadiationPhysics/picongpu/blob/0655b4426eac1122e23fb580315f867f7253c699/include/picongpu/particles/boundary/Utility.hpp#L97 

But, `getExternalCellsTotal` returns other values. For example for the `BACK` (or z-positive) boundary in a 256x256x128 simulation:
```
PhotonDetector: [direction9] beginExternalCellsTotal = 0, 0, 128
PhotonDetector: [direction9] endExternalCellsTotal = 256, 256, 128
```
so there are no cells in the valid z range. I think, it should be `0, 0, 128` and `256, 256, 128 + guarding_cells`

And for the FRONT boundary (z -neagtive):
```
PhotonDetector: [direction18] beginExternalCellsTotal = 0, 0, 0
PhotonDetector: [direction18] endExternalCellsTotal = 256, 256, 0
```
Here it should be `0, 0, 0` and `256, 256, guarding_cells`

@sbastrakov since this is your code, could you maybe have a look at this? I think with this the `ParticleCalorimeter` is not processing any outgoing particles.

**Comment by sbastrakov** [comment]
Thanks for reporting. This appears incorrect indeed, testing.

**Comment by sbastrakov** [comment]
This function, as all others in that file, operate with domains as defined [here](https://github.com/ComputationalRadiationPhysics/picongpu/wiki/PIConGPU-domain-definitions) and no guards. It is a bit messy that we combine both with- and without-guard indices as same data types and names in other parts and it gets mixed. 

I think the correct result of this function for `z-max` boundary should be
```
PhotonDetector: [direction9] beginExternalCellsTotal = -guarding_cells_x, -guarding_cells_y, 128 - boundaryOffset_z
PhotonDetector: [direction9] endExternalCellsTotal = 256 + guarding_cells_x, 256 + guarding_cells_y, 128 + guarding_cells_z
```
and for `z-min` boundary:
```
PhotonDetector: [direction18] beginExternalCellsTotal = -guarding_cells_x, -guarding_cells_y, -guarding_cells_z
PhotonDetector: [direction18] endExternalCellsTotal = 256 + guarding_cells_x, 256 + guarding_cells_y, boundaryOffset_z
```

I am making a fixing PR, and will also explicitly write that the indices are without guard.

**Comment by psychocoderHPC** [comment]
> I am making a fixing PR, and will also explicitly write that the indices are without guard.

@sbastrakov Do you applied already the fix to the mainline code base?

**Comment by sbastrakov** [comment]
This was fixed with #4030.
